charts/datadog/templates/helm-check-rbac.yaml

```diff
@@ -31,7 +31,7 @@ subjects:
     {{- if and .Values.datadog.clusterChecks.enabled .Values.clusterChecksRunner.enabled }}
     name: {{ if .Values.clusterChecksRunner.rbac.create }}{{ template "datadog.fullname" . }}-cluster-checks{{ else }}"{{ .Values.clusterChecksRunner.rbac.serviceAccountName }}"{{ end }}
     {{- else }}
-    name: {{ if .Values.clusterAgent.rbac.create }}{{ template "datadog.fullname" . }}-cluster-agent{{ else }}"{{ .Values.clusterAgent.rbac.serviceAccountName }}"{{ end }}
+    name: {{ include "agents.serviceAccountName" . }}
```

Comment:
This was supposed to be service account used by the cluster agent. But here, it is replaced by the service account used by the agents.

Reply:
right thanks for catching this

Reply:
```suggestion
    name: {{ if .Values.clusterAgent.rbac.create }}{{ template "datadog.fullname" . }}-cluster-agent{{ else }}"{{ .Values.clusterAgent.rbac.serviceAccountName }}"{{ end }}
```